jvm-packages/xgboost4j-gpu/src/test/java/ml/dmlc/xgboost4j/gpu/java/BoosterTest.java

```diff
@@ -83,7 +85,16 @@ public void testBooster() throws XGBoostError {
 
     try (Table tmpTable = Table.readCSV(schema, opts,
 					new File(trainingDataPath))) {
-      CudfColumnBatch batch = new CudfColumnBatch(tmpTable, new int[]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}, new int[]{12});
+      ColumnVector[] df = new ColumnVector[12];
+      for (int i = 0; i < 12; ++i) {
+        df[i] = tmpTable.getColumn(i);
+      }
+      Table X = new Table(df);
+      ColumnVector[] labels = new ColumnVector[1];
+      labels[0] = tmpTable.getColumn(12);
+      Table y = new Table(labels);
+
+      CudfColumnBatch batch = new CudfColumnBatch(X, y, null, null);
```

Comment:
What's the necessity of the `CudfColumnBatch` class? Could `DeviceQuantileDMatrix` directly accept this object `ColumnVector[] df`?

Reply:
DQDM accepts only iterators.  The `CudfColumnBatch` is an adapter.  I have made a similar suggestion to @wbo4958 so I'm open to consuming `Iterator<Table>` directly and move DQDM into xgboost4j-gpu package (so we don't bring cuDF into CPU package's dependency tree).

Reply:
Is it possible for DQDM to accept different iterators for labels and feature columns? e.g. `DeviceQuantileDMatrix(features.iterator(), labels.iterator(), Float.NaN, maxBin, 1);`

Reply:
But that also complicates the construction of normal DMatrix, which cannot have cudf table in its definition due to dependency issues.

Reply:
@RAMitchell 

> Is it possible for DQDM to accept different iterators for labels and feature columns?

Yes, but with some other caveats.  The first possible approach is we pass down all the iterators into libxgboost.so, but we will have a weird and inflexible C API that accepts many iterator pointers.  The second approach would be we concatenate all the iterators into one in java, but that's effectively the same as current implementation except that we do it for the users.